src/app/charts/core/files/secrets/tpl/spring-config-saml.properties

```diff
@@ -0,0 +1,45 @@
+{{- $samlconfig := (include "arkcase.saml" $) -}}
+{{- if $samlconfig -}}
+  {{- $conf := get (fromYaml $samlconfig) "saml" -}}
+  {{- if $conf -}}
+  {{- $baseUrl := get (fromYaml $samlconfig) "arkcaseUrl" -}}
+  {{- $arkcaseUrl := (include "arkcase.tools.parseUrl" $baseUrl) | fromYaml -}}
```

Comment:
These two lines can be consolidated into one:

`$arkcaseUrl := (include "arkcase.tools.parseUrl" $samlconfig.arkcaseUrl | fromYaml) -}}`

... assuming you do the `fromYaml` thing off the rip on line 1, like I suggested...

Reply:
Yes, this is all much cleaner now, thanks.

Reply:
You can also consider including the already-parsed URL in your computation of the SAML configuration object, so you don't have to do this step here and already have the final object available.